astropy/utils/masked/core.py

```diff
@@ -451,9 +549,11 @@ def __getitem__(self, indx):
         return self._masked.from_unmasked(out, mask, copy=False)
 
     def __setitem__(self, index, value):
-        data, mask = self._masked._get_data_and_mask(value, allow_ma_masked=True)
-        if data is not None:
-            self._dataiter[index] = data
+        if value is np.ma.masked or value is np.ma.nomask:
+            mask = value is np.ma.masked
+        else:
+            unmasked, mask = get_data_and_mask(value)
+            self._dataiter[index] = value
```

Comment:
Should this be `self._dataiter[index] = unmasked`?

Reply:
Yes, thanks! (It will actually do the same thing, but clearly I just forgot to rename the variable...)